src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -614,15 +578,15 @@ def _upad_input(self, query_layer, key_layer, value_layer, attention_mask, query
 
 
 class WhisperSdpaAttention(WhisperAttention):
-    # Copied from transformers.models.bart.modeling_bart.BartSdpaAttention.forward with BART->whisper, Bart->Whisper
     def forward(
         self,
         hidden_states: torch.Tensor,
         key_value_states: Optional[torch.Tensor] = None,
-        past_key_value: Optional[Tuple[torch.Tensor]] = None,
+        past_key_value: Optional[EncoderDecoderCache] = None,
```

Comment:
If we still suppport legacy it has to be Union Tuple EncoderDecoderCache

Reply:
Likewise here: https://github.com/huggingface/transformers/pull/31166#discussion_r1655043228